internal/presentation/gui.go

```diff
@@ -14,26 +14,75 @@ type gameUi struct {
 	loop *gameLoop
 
 	boardDrawer BoardDrawer
-	state       *model.Board
+	board       *model.Board
 }
 
-func (g gameUi) drawButton(name, text string, topLeftX, topLeftY, bottomRightX, bottomRightY int, handler GuiHandler) error {
+func (g gameUi) backgroundColor() gocui.Attribute {
+	switch g.board.CurrentPlayer().Color {
+	case model.ColorBlue:
+		return gocui.ColorBlue
+	case model.ColorGreen:
+		return gocui.ColorGreen
+	case model.ColorRed:
+		return gocui.ColorRed
+	default:
+		return gocui.ColorYellow
+	}
+}
+
+func (g gameUi) foregroundColor() gocui.Attribute {
+	switch g.board.CurrentPlayer().Color {
+	case model.ColorBlue, model.ColorGreen, model.ColorRed:
+		return gocui.ColorWhite
+	default:
+		return gocui.ColorBlack
+	}
+}
+
+func (g gameUi) drawButton(name, text string, topLeftX, topLeftY, bottomRightX, bottomRightY int, handler GuiHandler) (*gocui.View, error) {
```

Comment:
I thought the practice in go is to return the error first. Out of curiosity, why do you return the gocuiView first here?

Reply:
The error is the last parameter in idiomatic Go (See https://go.dev/blog/error-handling-and-go).